### PR TITLE
[BE] update wrapFunctionCalls

### PR DESCRIPTION
**Pull request by kabdelhak**
- ticket #5771
 - check noEvent arguments inside if condition

### CODE REVIEW

File: OMCompiler/Compiler/BackEnd/CommonSubExpression.mo [hunk 1]
```diff
@@ -885,11 +884,22 @@ algorithm
...
+    /*
+      check arguments of noEvent for functions but ignore noEvent itself.
+      ticket #5771
+    */
+    case DAE.CALL(Absyn.IDENT("noEvent"),{DAE.RELATION(e,_,e2,_,_)}) algorithm
+      (_, outTuple) := Expression.traverseExpTopDown(e, wrapFunctionCalls_analysis3, inTuple);
+      (_, outTuple) := Expression.traverseExpTopDown(e2, wrapFunctionCalls_analysis3, outTuple);
+      cont := false;
+      return;
+    then true;
```
Comment (lochel) [inline OMCompiler/Compiler/BackEnd/CommonSubExpression.mo]: @kabdelhak What does the return statement in this case?

Comment (kabdelhak) [reply OMCompiler/Compiler/BackEnd/CommonSubExpression.mo]: Hmmm nothing i guess. I just copied the behaviour from the DAE.IFEXP() case. I could just return false.

Comment (kabdelhak) [reply OMCompiler/Compiler/BackEnd/CommonSubExpression.mo]: Ah i see, i tried just returning false and it failed. The return statement prevents the last line in this function from being executed. ` outTuple := (HT, exarray, cseIndex, index, functionTree);`

Comment (lochel) [reply OMCompiler/Compiler/BackEnd/CommonSubExpression.mo]: How would it be different to `fail()`?

Comment (lochel) [reply OMCompiler/Compiler/BackEnd/CommonSubExpression.mo]: Ahh, it is not failing but returning with an undefined value for `outTuple`, right?

Comment (lochel) [reply OMCompiler/Compiler/BackEnd/CommonSubExpression.mo]: Never mind - now I got it!

Comment (kabdelhak) [reply OMCompiler/Compiler/BackEnd/CommonSubExpression.mo]: Ok perfect. :D

**Pull request merged**